src/transform-sdk/cpp/include/redpanda/transform_sdk.h

```diff
@@ -213,4 +234,190 @@ using on_record_written_callback
  */
 [[noreturn]] void on_record_written(const on_record_written_callback& callback);
 
+namespace sr {
+
+/**
+ * enum indicating the format of some schema
+ * Supported formats:
+ *   - avro (fully supported)
+ *   - protobuf (fully supported)
+ *   - json (redpanda v24.2+ WIP)
+ */
+enum class schema_format {
+    avro = 0,
+    protobuf = 1,
+    json = 2,
+};
+
+/**
+ * Type-safe wrapper for a the ID (int32) of some registered schema
+ */
+using schema_id = detail::simple_named_type<int32_t, struct schema_id_tag>;
+
+/**
+ * Type-safe wrapper for a bare schema version (int32)
+ */
+using schema_version
+  = detail::simple_named_type<int32_t, struct schema_version_tag>;
+
+/**
+ * Aggregate representing a reference to a schema defined elsewhere.
+ * The details are format dependent; e.g Avro schemas will use a different
+ * reference syntax from protobuf of JSON schemas. For more information , see:
+ *
+ *   https://docs.redpanda.com/current/manage/schema-reg/schema-reg-api/#reference-a-schema
+ */
+struct reference {
+    std::string name;
+    std::string subject;
+    schema_version version;
+
+private:
+    friend bool operator==(const reference&, const reference&) = default;
```

Comment:
is this to disable comparison?

Reply:
Nah, this should generate a default overload for the reference struct.

Per the standard, if it's not a member it must be a friend. For some "reason" i mildly prefer the friend formulation to the non-static member 🤷‍♂️

Reply:
Oh, the privateness? doesn't matter since it's not a member. Private friend feels natural to me for whatever reason